frontend/app/routes/_gcweb-app.letters._index.tsx

```diff
@@ -86,14 +73,18 @@ export default function LettersIndex() {
         />
       </div>
       <ul className="divide-y border-y">
-        {letters.map((letter) => (
-          <li key={letter.id} className="py-4 sm:py-6">
-            <Link reloadDocument to={`/letters/${letter.referenceId}/download`} className="font-medium hover:underline">
-              {i18n.language === 'en' ? letter.nameEn : letter.nameFr}
-            </Link>
-            <p className="mt-1 text-sm text-gray-500">{t('letters:index.date', { date: getFormattedDate(letter.dateSent) })}</p>
-          </li>
-        ))}
+        {letters.map((letter) => {
+          const letterType = letterTypes.find(({ code }) => code === letter.name);
+          const letterName = letterType ? (i18n.language === 'fr' ? letterType.nameFr : letterType.nameEn) : letter.name;
```

Comment:
```suggestion
          // prettier-ignore
          const letterName = letterType
            ? i18n.language === 'fr'
              ? letterType.nameFr
              : letterType.nameEn
            : letter.name;
```

Not sure if that's any more readable. I hate ternaries, and I hate² nested ternaries. Is there any way this can be cleaned up to be a little to read?

Reply:
``` typescript
// prettier-ignore
const letterName = letterType
  ? letterType[i18n.language === 'fr' ? 'nameFr' : 'nameEn']
  : letter.name;
```

??

Ternaries are gross when used for anything other than trivial either/ors